fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/debug/PlannerRepl.java

```diff
@@ -272,6 +287,9 @@ public void onEvent(final Event event) {
                         final Commands.Command<Event> command = commandOptional.get();
                         final Optional<Boolean> isContinueOptional = getSilently("run command", () -> command.executeCommand(this, event, parsedLine));
                         isContinue = isContinueOptional.orElse(false);
+                        if (command instanceof Commands.QuitCommand) {
```

Comment:
why not just communicate `isContinue` back to here and then let the real do its thing?

Reply:
That's even better, I will try that.

Reply:
hmm... it looks like `isContinueOptional` just indicates whether the planner should continue or prompt for the next command, i.e. it does not break the debugger main loop:

```java
/**
* ...
* @return {@code true} if planning should continue afterwards, {@code false} if the REPL should prompt
*         for the next command after the execution of this command has finished.
*/
boolean executeCommand(@Nonnull PlannerRepl plannerRepl, @Nonnull E event, @Nonnull ParsedLine parsedLine);
```

Reply:
That's all it ever does. There is no really breaking out other than `System.exit()`. You can throw an exception as an alternative.